service/controller/clusterapi/v29/resource/tcdp/create.go

```diff
@@ -2,22 +2,29 @@ package tcdp
 
 import (
 	"context"
+	"strconv"
+	"strings"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/cloudformation"
 	"github.com/giantswarm/microerror"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 
 	"github.com/giantswarm/aws-operator/service/controller/clusterapi/v29/controllercontext"
 	"github.com/giantswarm/aws-operator/service/controller/clusterapi/v29/key"
-	"github.com/giantswarm/aws-operator/service/controller/clusterapi/v29/resource/cpi/template"
+	"github.com/giantswarm/aws-operator/service/controller/clusterapi/v29/resource/tcdp/template"
 )
 
 const (
 	capabilityNamesIAM = "CAPABILITY_NAMED_IAM"
 )
 
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
-	cr, err := key.ToMachineDeployment(obj)
+	cl, err := r.toClusterFunc(obj)
```

Comment:
I find it weird that the cluster type is necessary here but it does not easily work without it now.

Reply:
Sure, we can improve things like this with iteration.